### DISCUSSION TITLE
post_install hook fails when using poetry --directory

### DISCUSSION THREAD

**Discussion by worldworm** [nat-n/poethepoet#209] [Q&A] [answered]
Hi there,

I stumbled upon a weird thing. I'm not sure if this is a bug, fits into the [known limitations](https://poethepoet.natn.io/poetry_plugin.html#known-limitations), or I'm using it completely wrong.

I've got a fullstack monorepo with the following structure:

```
project/
    .git/
    .pre-commit-config.yaml
    frontend/
        package.json
    backend/
        pyproject.toml
```
The structure is generated via a [copier](https://github.com/copier-org/copier) template. After generating it automatically installs the npm dependencies `npm --prefix frontend install` and also the backend dependencies with `poetry --directory backend install`

Note: it will not cd into the corresponding directories and will execute the commands from the `project` folder.

When i now install a post_install hook like this:
```toml
[tool.poe.poetry_hooks]
post_install = "pre-commit-install"

[tool.poe.tasks]
pre-commit-install = "pre-commit install --install-hooks"
```
It will beautifully work if I'm in the `project` dir and call `poe --root backend pre-commit-install`. It will also work if I'm running `poetry install` from `project/backend/`

**BUT** running `poetry --directory backend install` will not work and fails with the follwing message:
`Error: Poe could not find a pyproject.toml file in /project or its parents`

For me it looks like the `--directory` is not "translated" correctly into the `--root` or poethepoet is not aware of this.

---

I know that I am misusing the functionality a bit here. Also it is a bit buggy because the pre-commit config is outside the poetry folder structure.
Do you have any ideas?
(Otherwise, I would not use a pre_install hook and simply install it manually, but i love the hook as it forces the user to install pre-commit if `poetry install` is run manually 😄 )

**Comment by nat-n** [comment, marked as answer]
I think this might be a bug. I suspect the issue is that the poethepoet plugin naively assumes it should inherit the CWD from the poetry process here: https://github.com/nat-n/poethepoet/blob/main/poethepoet/plugin.py#L59 

There is probably a way for the plugin to get from poetry that it is being run with a different directory (or rather the location of the pyproject.toml used by poetry). It'll take some fiddling to work out how. I suspect the same failure mode will occur when running a poe task directly with `poetry --directory ...` :(

I'll add this to my todo list, but if you want a fix soon any help working out exactly how to do this would be appreciated.

**Reply by worldworm** [reply]
Thank you for pointing me in the right direction.
Have a look here https://github.com/nat-n/poethepoet/pull/211 😃
